### ISSUE TITLE
Build error ssids on Mac with Intel compiler

### ISSUE THREAD

**Issue by amontoison** [ralna/GALAHAD#81]
@jfowkes 
```
 FAILED: libgalahad_single.dylib.p/src_ssids_ldlt_app.cxx.o 
c++ -Ilibgalahad_single.dylib.p -I. -I.. -I../include -fcolor-diagnostics -Wall -Winvalid-pch -std=c++14 -O0 -g -DGALAHAD_SINGLE -DSPRAL_SINGLE -DSPRAL_NO_HWLOC -DSPRAL_HAVE_SCHED_GETCPU -MD -MQ libgalahad_single.dylib.p/src_ssids_ldlt_app.cxx.o -MF libgalahad_single.dylib.p/src_ssids_ldlt_app.cxx.o.d -o libgalahad_single.dylib.p/src_ssids_ldlt_app.cxx.o -c ../src/ssids/ldlt_app.cxx
In file included from ../src/ssids/ldlt_app.cxx:6:
In file included from ../include/ssids_cpu_kernels_ldlt_app.hxx:8:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/vector:275:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__bit_reference:15:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/algorithm:651:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/cstddef:37:
../version:1:3: error: C++ requires a type specifier for all declarations
* GALAHAD version 4.0.1 (4/April/2022 14:10 GMT)
  ^
../version:1:10: error: expected ';' after top level declarator
* GALAHAD version 4.0.1 (4/April/2022 14:10 GMT)
         ^
In file included from ../src/ssids/ldlt_app.cxx:6:
In file included from ../include/ssids_cpu_kernels_ldlt_app.hxx:8:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/vector:275:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__bit_reference:15:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/algorithm:653:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/functional:490:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__algorithm/search.h:15:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__iterator/iterator_traits.h:14:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__iterator/incrementable_traits.h:14:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/concepts:134:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__functional_base:22:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/exception:83:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/cstdlib:85:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/stdlib.h:93:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/stdlib.h:66:
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/wait.h:251:16: error: unknown type name 'idtype_t'; did you mean 'wctype_t'?
int     waitid(idtype_t, id_t, siginfo_t *, int) __DARWIN_ALIAS_C(waitid);
               ^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/_types/_wctype_t.h:32:27: note: 'wctype_t' declared here
typedef __darwin_wctype_t wctype_t;
                          ^
In file included from ../src/ssids/ldlt_app.cxx:6:
In file included from ../include/ssids_cpu_kernels_ldlt_app.hxx:8:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/vector:275:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__bit_reference:15:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/algorithm:653:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/functional:490:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__algorithm/search.h:15:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__iterator/iterator_traits.h:14:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__iterator/incrementable_traits.h:14:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/concepts:134:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__functional_base:22:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/exception:85:
../version:1:3: error: C++ requires a type specifier for all declarations
* GALAHAD version 4.0.1 (4/April/2022 14:10 GMT)
  ^
../version:1:10: error: expected ';' after top level declarator
* GALAHAD version 4.0.1 (4/April/2022 14:10 GMT)
         ^
In file included from ../src/ssids/ldlt_app.cxx:6:
In file included from ../include/ssids_cpu_kernels_ldlt_app.hxx:8:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/vector:275:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__bit_reference:15:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/algorithm:653:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/functional:490:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__algorithm/search.h:15:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__iterator/iterator_traits.h:14:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__iterator/incrementable_traits.h:14:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/concepts:134:
In file included from /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__functional_base:23:
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/new:181:108: error: no type named 'nothrow_t' in namespace 'std'
_LIBCPP_NODISCARD_AFTER_CXX17 _LIBCPP_OVERRIDABLE_FUNC_VIS void* operator new(std::size_t __sz, const std::nothrow_t&) _NOEXCEPT _LIBCPP_NOALIAS;
                                                                                                      ~~~~~^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/new:183:74: error: no type named 'nothrow_t' in namespace 'std'
_LIBCPP_OVERRIDABLE_FUNC_VIS void  operator delete(void* __p, const std::nothrow_t&) _NOEXCEPT;
                                                                    ~~~~~^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/new:189:110: error: no type named 'nothrow_t' in namespace 'std'
_LIBCPP_NODISCARD_AFTER_CXX17 _LIBCPP_OVERRIDABLE_FUNC_VIS void* operator new[](std::size_t __sz, const std::nothrow_t&) _NOEXCEPT _LIBCPP_NOALIAS;
                                                                                                        ~~~~~^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/new:191:76: error: no type named 'nothrow_t' in namespace 'std'
_LIBCPP_OVERRIDABLE_FUNC_VIS void  operator delete[](void* __p, const std::nothrow_t&) _NOEXCEPT;
                                                                      ~~~~~^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/new:197:102: error: no type named 'align_val_t' in namespace 'std'
_LIBCPP_NODISCARD_AFTER_CXX17 _LIBCPP_OVERRIDABLE_FUNC_VIS void* operator new(std::size_t __sz, std::align_val_t) _THROW_BAD_ALLOC;
                                                                                                ~~~~~^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/new:198:102: error: no type named 'align_val_t' in namespace 'std'
_LIBCPP_NODISCARD_AFTER_CXX17 _LIBCPP_OVERRIDABLE_FUNC_VIS void* operator new(std::size_t __sz, std::align_val_t, const std::nothrow_t&) _NOEXCEPT _LIBCPP_NOALIAS;
                                                                                                ~~~~~^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/new:198:126: error: no type named 'nothrow_t' in namespace 'std'
_LIBCPP_NODISCARD_AFTER_CXX17 _LIBCPP_OVERRIDABLE_FUNC_VIS void* operator new(std::size_t __sz, std::align_val_t, const std::nothrow_t&) _NOEXCEPT _LIBCPP_NOALIAS;
                                                                                                                        ~~~~~^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/new:199:68: error: no type named 'align_val_t' in namespace 'std'
_LIBCPP_OVERRIDABLE_FUNC_VIS void  operator delete(void* __p, std::align_val_t) _NOEXCEPT;
                                                              ~~~~~^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/new:200:68: error: no type named 'align_val_t' in namespace 'std'
_LIBCPP_OVERRIDABLE_FUNC_VIS void  operator delete(void* __p, std::align_val_t, const std::nothrow_t&) _NOEXCEPT;
                                                              ~~~~~^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/new:200:92: error: no type named 'nothrow_t' in namespace 'std'
_LIBCPP_OVERRIDABLE_FUNC_VIS void  operator delete(void* __p, std::align_val_t, const std::nothrow_t&) _NOEXCEPT;
                                                                                      ~~~~~^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/new:202:124: error: no type named 'align_val_t' in namespace 'std'
_LIBCPP_OVERRIDABLE_FUNC_VIS _LIBCPP_AVAILABILITY_SIZED_NEW_DELETE void  operator delete(void* __p, std::size_t __sz, std::align_val_t) _NOEXCEPT;
                                                                                                                      ~~~~~^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/new:205:104: error: no type named 'align_val_t' in namespace 'std'
_LIBCPP_NODISCARD_AFTER_CXX17 _LIBCPP_OVERRIDABLE_FUNC_VIS void* operator new[](std::size_t __sz, std::align_val_t) _THROW_BAD_ALLOC;
                                                                                                  ~~~~~^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/new:206:104: error: no type named 'align_val_t' in namespace 'std'
_LIBCPP_NODISCARD_AFTER_CXX17 _LIBCPP_OVERRIDABLE_FUNC_VIS void* operator new[](std::size_t __sz, std::align_val_t, const std::nothrow_t&) _NOEXCEPT _LIBCPP_NOALIAS;
                                                                                                  ~~~~~^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/new:206:128: error: no type named 'nothrow_t' in namespace 'std'
_LIBCPP_NODISCARD_AFTER_CXX17 _LIBCPP_OVERRIDABLE_FUNC_VIS void* operator new[](std::size_t __sz, std::align_val_t, const std::nothrow_t&) _NOEXCEPT _LIBCPP_NOALIAS;
                                                                                                                          ~~~~~^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated
```

**Comment by nimgould** [comment]
I hope someone on the spral side can help here ... we don't have any native C++ expertise, but external people sometimes help. @jfowkes may be able to ask our spral contributors. Can this check not be turned off in the compiler?

**Comment by jfowkes** [comment]
@amontoison you're missing `-std=c++17` from the C++ flags, this does not seem to be assumed by default on macOS.

**Comment by jfowkes** [comment]
@amontoison looks like the macOS build is not propagating the C++17 support for some reason, `align_val_t`  has been in the standard namespace since C++17:
https://en.cppreference.com/w/cpp/memory/new/align_val_t

**Comment by amontoison** [comment]
@jfowkes, I added `-std=c++17` but I still have some issues: https://github.com/ralna/GALAHAD/actions/runs/4108591916/jobs/7089458310#step:8:52
Do you have an idea how to solve them?

**Comment by jfowkes** [comment]
@amontoison two issues with `standalone.yml`:

1.  `modflowpy/install-gfortran-action@v1` should only be installed when compiling with gfortran, currently it is always installed!!
2. `modflowpy/install-intelfortran-action@v1` should be used to install the Intel OneAPI compilers

SPRAL on macOS with Intel builds fine for us as you can see in https://github.com/ralna/spral/pull/97

**Comment by amontoison** [comment]
I can see that MacOS + Intel compilers is failing Jari https://github.com/ralna/spral/pull/97

**Comment by amontoison** [comment]
1- Yes, I know but because I always specify the compilers to Meson with `CC`, `CXX` and `FC` it's not a problem. I will   update the workflow to fix it.
2- I didn't that the action `modflowpy/install-intelfortran-action@v1` was available. I will test it.

**Comment by jfowkes** [comment]
> I can see that MacOS + Intel compilers is failing Jari [ralna/spral#97](https://github.com/ralna/spral/pull/97)

@amontoison yes but on the SSMFE test, SSIDS builds and passes all tests.

**Comment by amontoison** [comment]
Fixed with #118
